### PR TITLE
Refactor and add tests

### PR DESCRIPTION
**Pull request by tcbyrd**
This doesn't change or fix anything. I started with just adding tests then refactored a bit to make the tests make a little more sense. Going to test this branch against a real pages site to verify it works.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -147,8 +133,20 @@ async function cancelHandler(evtOrExitCodeOrError) {
...
+    await deployment.create()
+    await deployment.check()
+  } catch (error) {
+    core.setFailed(error)
+  }
+}
+
+process.on('SIGINT', cancelHandler)
+
+process.on('SIGTERM', cancelHandler)
```
Comment (tcbyrd) [inline src/index.js:148]: @YiMysty es linter really didn't want us to do it your way. Kept adding a semicolon at the beginning of the line, so I just broke it out this way.

File: src/index.js [hunk 2]
```diff
@@ -147,8 +133,20 @@ async function cancelHandler(evtOrExitCodeOrError) {
...
 }
 
-[
-  'SIGINT',  'SIGTERM',
-].forEach(evt => process.on(evt, cancelHandler))
+async function main() {
+  try {
+    const deployment = new Deployment()
+    await deployment.create()
+    await deployment.check()
```
Comment (tcbyrd) [inline src/index.js:140]: This seems weird for now. Will break this up into separate error handlers for better debugging in a follow up. Mainly trying to make sure this all works and get CI in a state where we can trust it.

**Review by YiMysty — APPROVED** [review]

**Comment by tcbyrd** [comment]
Use this branch here and it deployed successfully! https://github.com/paper-spa/deploy-pages-test/runs/4245106821?check_suite_focus=true#step:2:1

**Review by jcudit — APPROVED** [review]
Looking forward to this data 🙇🏾

**Pull request merged**